src/core/thread/mle.cpp

```diff
@@ -1724,7 +1726,11 @@ ThreadError Mle::HandleDataResponse(const Message &aMessage, const Ip6::MessageI
     // Active Timestamp
     if (Tlv::GetTlv(aMessage, Tlv::kActiveTimestamp, sizeof(activeTimestamp), activeTimestamp) == kThreadError_None)
     {
+        const MeshCoP::Dataset local = mNetif.GetActiveDataset().GetLocal();
+        const MeshCoP::Timestamp *localTimestamp = local.GetTimestamp();
+
         VerifyOrExit(activeTimestamp.IsValid(), error = kThreadError_Parse);
+        VerifyOrExit(update || (localTimestamp && (activeTimestamp.Compare(*localTimestamp) == 0)), ;);
```

Comment:
In spec section 5.15.3, if the incoming message also includes an MLE Network Data TLV and contains Active Timestamp TLV that matches local Active Timestamp.

It asked to be matched. And Set function update while timestamp is greater. So I compare it here.